File: frontends/telegram/handlers/commands.py
"""Command handlers for the Notes Bot."""

import logging
from telegram import Update
from telegram.ext import ContextTypes

from ..config import ROOT_ID
from ..keyboards.main_menu import get_main_menu_keyboard
from ..states import state_manager
from ..utils import escape_markdown_v2

logger = logging.getLogger(__name__)


async def cmd_start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    if not update.message or not update.effective_user:
        return

    user_id = update.effective_user.id

    if ROOT_ID and user_id != ROOT_ID:
        await update.message.reply_text("⛔ Unauthorized access.")
        logger.warning(f"Unauthorized access attempt from user {user_id}")
        return

    active_date = state_manager.get_context(user_id).active_date

    await update.message.reply_text(
        f"👋 Добро пожаловать\\!\n\n"
        f"📅 Активная дата: {escape_markdown_v2(active_date)}\n\n"
        f"Выберите действие:",
        reply_markup=get_main_menu_keyboard(active_date),
        parse_mode="MarkdownV2",
    )
    logger.info(f"User {user_id} started the bot")

File: frontends/telegram/tghandlers/commands.go
package tghandlers

import (
	"context"
	"fmt"

	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
	"go.uber.org/zap"

	"notes-bot/frontends/telegram/tgfmt"
	"notes-bot/internal/telemetry"
)

func (a *App) HandleStart(ctx context.Context, tgBot *tgbotapi.BotAPI, update *tgbotapi.Update) {
	ctx, span := telemetry.StartSpan(ctx)
	defer span.End()

	if update.Message == nil || update.Message.From == nil {
		return
	}

	userID := update.Message.From.ID
	if !a.authorized(userID) {
		sendText(ctx, tgBot, update.Message.Chat.ID, tgfmt.Escape("⛔ Unauthorized access."), nil, true)
		a.Logger.Warn("unauthorized access attempt", zap.Int64("user_id", userID))
		return
	}

	uc, err := a.State.GetContext(ctx, userID)
	if err != nil {
		a.Logger.Error("get context", zap.Error(err))
		return
	}

	text := tgfmt.Join(
		tgfmt.Escape("👋 Добро пожаловать!\n\n📅 Активная дата: "),
		tgfmt.Code(tgfmt.Escape(fmt.Sprintf("%s", uc.ActiveDate))),
		tgfmt.Escape("\n\nВыберите действие:"),
	)
	kb := a.getMainMenuKeyboard(ctx)
	if err := replyToUpdate(ctx, tgBot, update, text, &kb); err != nil {
		a.Logger.Error("send start message", zap.Error(err))
	}
	a.Logger.Info("user started bot", zap.Int64("user_id", userID))
}
